src/components/UserAccess/__tests__/user-access-actions.spec.ts

```diff
@@ -1,100 +1,111 @@
-import '@testing-library/jest-dom';
-import { renderHook } from '@testing-library/react-hooks';
+import { mockRoleBinding, mockRoleBindingWithoutUser } from '../../../__data__/rolebinding-data';
 import { useAccessReviewForModel } from '../../../utils/rbac';
-import { createUseWorkspaceInfoMock } from '../../../utils/test-utils';
-import { useSBRActions } from '../user-access-actions';
+import { useModalLauncher } from '../../modal/ModalProvider';
+import { useWorkspaceInfo } from '../../Workspace/useWorkspaceInfo';
+import { useRBActions } from '../user-access-actions';
 
+// Mock the dependencies
 jest.mock('../../../utils/rbac', () => ({
   useAccessReviewForModel: jest.fn(),
 }));
 
-const accessReviewMock = useAccessReviewForModel as jest.Mock;
+jest.mock('../../Workspace/useWorkspaceInfo', () => ({
+  useWorkspaceInfo: jest.fn(),
+}));
+
+jest.mock('../../modal/ModalProvider', () => ({
+  useModalLauncher: jest.fn(),
+}));
 
-describe('useSBRActions', () => {
-  createUseWorkspaceInfoMock({ workspace: 'test-ws' });
+describe('useRBActions', () => {
+  const mockWorkspace = 'test-workspace';
 
   beforeEach(() => {
-    accessReviewMock.mockReturnValue([true, true]);
+    // Reset mock functions before each test
+    (useAccessReviewForModel as jest.Mock).mockClear();
+    (useWorkspaceInfo as jest.Mock).mockClear();
+    (useModalLauncher as jest.Mock).mockClear();
   });
 
-  it('should return enabled actions', () => {
-    const { result } = renderHook(() =>
-      useSBRActions({
-        availableActions: ['update', 'delete'],
-        masterUserRecord: 'my-user',
-        role: 'admin',
-        bindingRequest: { name: 'my-sbr', namespace: 'my-ns' },
-      }),
-    );
-    const actions = result.current;
+  it('should return Edit and Revoke actions when the user has permissions', () => {
+    (useAccessReviewForModel as jest.Mock).mockReturnValue([true, true]);
+    (useWorkspaceInfo as jest.Mock).mockReturnValue({ workspace: mockWorkspace });
+    (useModalLauncher as jest.Mock).mockReturnValue(jest.fn());
```

Comment:
Why do you need to mock the modal Launcher hook? It just returns a callback. As long as a callback is not called, mocking would not be needed

Reply:
Removed